project_and_code_guidelines.md

```diff
@@ -404,48 +401,102 @@ static final String ACTION_OPEN_USER = "com.myapp.action.ACTION_OPEN_USER";
 
 ### 2.2.14 Arguments in Fragments and Activities
 
-When data is passed into an Activity or Fragment via `Intents` or a `Bundles`, the keys for the different values __must__ follow the rules described in the section above. In addition, the constant should be defined in the `Activity` or `Fragment` class as `public`. 
+When data is passed into an `Activity `or `Fragment` via `Intents` or a `Bundles`, the keys for the different values __must__ follow the rules described in the section above.
 
-It's also a good practise to add a comment on top of the class to explain what values are expected and whether they are optional or required. 
+When an `Activity` or `Fragment` expect arguments, it should provide a `static public` method that facilitates the creation of the `Fragment` or `Intent`.
+
+In the case of Activities the method is usually called `getStartIntent()`
 
 ```java
-// This fragments requires an argument with key KEY_ARG_USER and type User
-public class UserFragment extends Fragment {
+public static Intent getStartIntent(Context context, User user) {
+	Intent intent = new Intent(context, ThisActivity.class);
+	intent.putParcelableExtra(EXTRA_USER, user);
+	return intent;
+}
+```
 
-    public static final String KEY_ARG_USER = "KEY_ARG_USER";
+For Fragments it's named `newInstance()` and it handles the creation of the Fragment with the right arguments. 
 
-	public onCreate() {
-	    Bundle args = getArguments();
-	    User user = args.getParcelable(KEY_ARG_USER);
-	}
+```java
+public static UserFragment newInstance(User user) {
+	UserFragment fragment = new UserFragment;
+	Bundle args = new Bundle();
+	args.putParcelable(ARGUMENT_USER, user);
+	fragment.setArguments(args)
+	return fragment;
 }
 ```
-	
-And this is how to pass the User object into the Fragment:
+
+__Note__ these methods should go at the top of the class before `onCreate()`
+
+### 2.2.15 Line length limit
+
+Code lines should not exceed __100 characters__. If the line is longer than this limit there are usually two options to reduce its length:
+
+* Extract a local variable or method (Preferable).
```

Comment:
Hmm, I disagree with Google there, but I'm not quite sure I have the authority to do so. 

Any memory allocation will have an impact, and it all builds up, especially on lower end hardware like the wonderful HTC Desire C ;)